### PR TITLE
docs: update quickstart deployments to config v2

### PR DESCRIPTION
**Pull request by davecheney**
Update quickstart docs and manifests for the change to v2 config.

Signed-off-by: Dave Cheney <dave@cheney.net>

### CODE REVIEW

File: deployment/render/daemonset-rbac.yaml [hunk 1]
```diff
@@ -37,7 +37,7 @@ spec:
         - containerPort: 8080
           name: http
```
Comment (cmaloney) [inline deployment/render/daemonset-rbac.yaml]: missing port 8443 here?

Comment (davecheney) [reply deployment/render/daemonset-rbac.yaml]: Yup. Good catch. I'll fix that.

File: docs/deploy-options.md [hunk 2]
```diff
@@ -14,8 +14,8 @@ In either case, a Service of `type: LoadBalancer` is set up to forward to the Co
 ## Install
 
 - Clone or fork the repository.
-- To install the DaemonSet, navigate to the `deployment/ds-json-v1` directory. OR
-- To install the Deployment, navigate to the `deployment/deployment-json-v1`.
+- To install the DaemonSet, navigate to the `deployment/ds-grpc-v2` directory. OR
+- To install the Deployment, navigate to the `deployment/deployment-grpc-v2`.
```
Comment (Bradamant3) [inline docs/deploy-options.md]: add "directory", thus:
To install the Deployment, navigate to the `deployment/deployment-grpc-v2` directory.

File: deployment/common/service.yaml [hunk 3]
```diff
@@ -4,15 +4,24 @@ metadata:
...
-  # should be a no-op.
-  service.beta.kubernetes.io/aws-load-balancer-backend-protocol: http
+  # This annotation puts the AWS ELB into "TCP" mode so that it does not
+  # to http negotiation at the ELB edge. This means the remote address of
+  # all connectinos will appear to be the internal address of the ELB.
+  service.beta.kubernetes.io/aws-load-balancer-backend-protocol: tcp
+  # To recover the remote ip 
+  # 1. Add the --use-proxy-protocol flag to your deployment or daemonset's contour serve command.
+  # 2. uncomment this line to enable PROXY protocol # on the ELB
+  # service.beta.kubernetes.io/aws-load-balancer-proxy-protocol: "*"
```
Comment (Bradamant3) [inline deployment/common/service.yaml]: Not sure where to put this comment, so duplicating in what seem to be the two obvious places.
Edit first comment (starting at l7):

# This annotation configures the AWS ELB for TLS passthrough.
# This means that the remote address of each connection appears to be
# the internal address of the ELB.
# To recover the external address:
# 1. Add the --use-proxy-protocol flag to the command to retrieve the address.
# 2. Uncomment the following line to enable the proxy protocol on the ELB.

Do we also want to add a comment line after the # service.beta ... line to point them to the docs for more info? Or is that getting too carried away?

Comment (jbeda) [reply deployment/common/service.yaml]: Having a docs page on the proxy protocol makes sense.  This is enough of a topic we probably want to delve in a bit and have pointers to the ELB docs and such.

Also worth calling out that this isn't needed if NLB is used.  That is a new feature in 1.9.

Comment (jbeda) [reply deployment/common/service.yaml]: We could also have another version of this service (aws-service.yaml) and another set of links to pull it all together.  At the end of the day this is screaming for ksonnet. :/

File: deployment/deployment-grpc-v2/02-service.yaml [hunk 4]
```diff
@@ -4,11 +4,14 @@ metadata:
  name: contour
  namespace: heptio-contour
  annotations:
-  # This annotation puts the AWS ELB into "TCP" mode so that it does not 
-  # to http negotiation at the ELB edge. This means the remote address of 
+  # This annotation puts the AWS ELB into "TCP" mode so that it does not
+  # to http negotiation at the ELB edge. This means the remote address of
```
Comment (jbeda) [inline deployment/deployment-grpc-v2/02-service.yaml]: s/to/do/

File: deployment/render/daemonset-norbac.yaml [hunk 5]
```diff
@@ -68,15 +68,24 @@ metadata:
  name: contour
  namespace: heptio-contour
  annotations:
-  # This annotation puts the AWS ELB into "HTTP" mode so that it adds the
-  # `X-Forwarded-For` header on the request. If you aren't running on AWS this
-  # should be a no-op.
-  service.beta.kubernetes.io/aws-load-balancer-backend-protocol: http
+  # This annotation puts the AWS ELB into "TCP" mode so that it does not
+  # to http negotiation at the ELB edge. This means the remote address of
```
Comment (jbeda) [inline deployment/render/daemonset-norbac.yaml]: s/to/do/

Make sure to re-render after updating.

File: deployment/deployment-grpc-v2/02-service.yaml (hunk 4 above)
Comment (jbeda) [reply deployment/deployment-grpc-v2/02-service.yaml]: Should this file be a soft link to common/services.yaml?

File: docs/deploy-options.md (hunk 2 above)
Comment (jbeda) [reply docs/deploy-options.md]: Document/talk about the AWS/TCP/Proxy situation?

**Comment by Bradamant3** [comment]
See #159 for AWS/TCP/Proxy doc content.

File: deployment/deployment-grpc-v2/02-service.yaml (hunk 4 above)
Comment (davecheney) [reply deployment/deployment-grpc-v2/02-service.yaml]: It is, or that is it say, it should be. I'll double check.

File: deployment/common/service.yaml (hunk 3 above)
Comment (davecheney) [reply deployment/common/service.yaml]: I think pulling this out into a proxy-protocol.md is the best solution. Unlike the elb mode, it won't be transparent on other clusters.

**Comment by Bradamant3** [comment]
I approved changes so as to unblock merging, but the generated files still need to be regenerated. And I'd like to copyedit the proxy doc together with the rest of the new doc.

**Pull request merged**